nautobot/dcim/forms.py

```diff
@@ -3235,8 +3232,17 @@ def __init__(self, *args, **kwargs):
             # Limit VLAN choices by Location
             if locations.count() == 1:
                 location = locations.first()
-                self.fields["untagged_vlan"].widget.add_query_param("locations", location.pk)
+                # In the case of a single location, use the available_on_device query param to limit untagged VLAN choices
+                # to those available on the devices in that location and in the ancestors of the location.
+                device_pks = [device.pk for device in devices]
+                self.fields["untagged_vlan"].widget.add_query_param("available_on_device", device_pks)
```

Comment:
Is this the right logical check to add this query parameter, or should it be in the `if device_count == 1` case below?

Reply:
I believe so as I understand it from this comment https://github.com/nautobot/nautobot/issues/6470#issuecomment-2546380761

Reply:
On review, I think this is OK - `available_on_device` basically translates to `(locations__isnull=True | locations__in=[device.location, device.location.ancestors for device in filter])`, so if the selected devices share the same location then `available_on_device` will evaluate the same as if a single device were selected.

Reply:
As such, it might not even be necessary to pass `device_pks` to the filter in this specific case, it might actually result in a simpler/more efficient query if you just pass devices.first().pk instead? Worth testing.

Reply:
I see cause we already checked that we only have one location. That makes sense